shell/protoc.sh

```diff
@@ -16,86 +11,166 @@ source "$SCRIPTS_DIR/lib/bootstrap.sh"
 
 PROTO_DOCS_DIR="$(get_repo_directory)/apidocs/proto"
 
-if [[ -n $CIRCLECI ]]; then
-  {
-    echo "warning: running protobuf generation in CI is only supported for bootstrap and is DEPRECATED"
-    echo "         this will result in only Go protobuf artifacts being generated"
-  } >&2
-  exec protoc --go_out=plugins=grpc,paths=source_relative:. --proto_path "$(get_repo_directory)/api" "$(get_repo_directory)/api/"*.proto
-fi
+PROTOC_IMPORTS_BASE_PATH="$HOME/.outreach/.protoc-imports"
+mkdir -p "$PROTOC_IMPORTS_BASE_PATH"
+
+info "Ensuring protoc imports are avaliable locally"
+
+imports="$(get_list "go-protoc-imports")"
 
-# Fallback if uid/gid is somehow empty
-if [[ -z $uid ]] || [[ -z $gid ]]; then
-  echo "Error: Failed to determine the uid/gid of the current user. Defaulting to standard 1000." >&2
-  uid=1000
-  gid=1000
+# Add default imports so they can get dealt with in the following loop just like custom imports.
+if has_feature "validation"; then
+  imports="${imports}{\"module\":\"github.com/envoyproxy/protoc-gen-validate@v$(get_application_version "protoc-gen-validate")\",\"path\":\"\"}$'\n'"
 fi
 
-get_import_basename() {
-  # shellcheck disable=SC2155 # Why: splitting declartion is messier
-  local module="$(jq -r .module <<<"$1")"
-  # sed removes the version off the module path if present
-  basename "$(go list -f '{{ .Path }}' -m "$module" | sed 's|/v[0-9]||')"
-}
-get_import_path() {
-  # shellcheck disable=SC2155 # Why: splitting declartion is messier
-  local module="$(jq -r .module <<<"$1")"
-  # sed removes the version off the module path if present
-  module_root="$(go list -f '{{ .Dir }}' -m "$module" | sed 's|/v[0-9]||')"
-  echo "${module_root}$(jq -r .path <<<"$1")"
-}
-for import in $(get_list "go-protoc-imports"); do
-  module=$(jq -r .module <<<"$import")
-  currentver="$(go version | awk '{ print $3 }' | sed 's|go||')"
-  requiredver="1.16.0"
-  if [ ! "$(printf '%s\n' "$requiredver" "$currentver" | sort -V | head -n1)" = "$requiredver" ]; then
-    echo "Go version must be greater than ${requiredver} to use 'go-protoc-imports' feature"
-    exit 1
+default_args=()
+for import in $imports; do
+  module_version_str=$(jq -r .module <<<"$import")
+  info_sub "$module_version_str"
+
+  # shellcheck disable=SC2206
+  module_version_arr=(${module_version_str//@/ })
+  module="${module_version_arr[0]}"
+  version="latest"
```

Comment:
Is this `latest` check here a change?  I'd really like protoc to be consistent run to run and not depend on any environment at all.  Can we fail if there isn't a version specified?

Reply:
looking at the old code I think it did default to latest because it was using `go install`, but we can definitely error on an unpinned version. since we're releasing this on the 15th of june which is already a breaking change release day this should be fine